### PR TITLE
fix #1595

### LINKED ISSUE #1595: 路由有问题
如果定义了两个路由 /xxx/:id     /xxx  访问/xxx 会进去第一个定义的路由/xxx/:id

### PR DESCRIPTION
**Pull request by ysqi**

### CODE REVIEW

File: tree.go [hunk 1]
```diff
@@ -265,15 +265,14 @@ func (t *Tree) addseg(segments []string, route interface{}, wildcards []string,
 			}
 			t.wildcard.addseg(segments[1:], route, append(wildcards, params...), reg+regexpStr)
 		} else {
-			var ok bool
 			var subTree *Tree
-			for _, subTree = range t.fixrouters {
-				if t.prefix == seg {
-					ok = true
+			for _, sub := range t.fixrouters {
```
Comment (JessonChan) [inline tree.go]: the code is more readable,but i can not find any clue to #1595. i think there's no difference,or there's is what i  have ignored?

Comment (ysqi) [inline tree.go]: find condition changed.
`t.prefix==seg` ==> `sub.prefix == seg`

Comment (JessonChan) [inline tree.go]: got it

**Comment by miraclesu** [comment]
LGTM

**Pull request merged**